### PR TITLE
Improved Quiz Selection & New version of the Blockchain game

### PR DESCRIPTION
**Pull request by larmork**
**Improvement of random selection in Quiz game:** now the questions are generated randomly, but not repeated, when the questions end, the user can restart the quiz.
**Improvement of the Blockchain game:** now the assembled puzzle has become colored, it is now possible to cancel the selection by double-clicking, a hint button and a game restart button have also appeared

### CODE REVIEW

File: src/components/App.tsx [hunk 1]
```diff
@@ -43,6 +43,7 @@ function App() {
             <Route path="/buy" element={<Buy />} />
             <Route path="/download/:id" element={<LinksRouter />} />
             <Route path="*" element={<NotFound />} />
+            <Route path="/blockchain" element={<Blockchain />} />
```
Comment (dgusakov) [inline src/components/App.tsx:46]: What is the purpose of the route?

File: src/components/Question.tsx [hunk 2]
```diff
@@ -1,26 +1,83 @@
...
+          <div className="container mt-2" id="back">
+            <div className="row">
+              <BackButton name="к темам" />
+            </div>
+          </div>
+          <div className="container mt-2">
+            <HeaderBold name={topic.displayName} />
+            <HeaderLight name="Вопрос" />
+            <div className="mt-3" id="question">
+              <div id="questionBody">Все вопросы из этой темы уже показаны.</div>
```
Comment (dgusakov) [inline src/components/Question.tsx:72]: We heed a "start over" button here

Comment (dgusakov) [reply src/components/Question.tsx:72]: Or I didn't get the idea. Is it a placeholder for the cases of no questions at all? Then change text

File: src/components/QuizMain.tsx [hunk 3]
```diff
@@ -1,23 +1,25 @@
...
+        <HeaderBoldCenter name="Выберите игру" /> {/* Header with bold styling and centered alignment */}
       </div>
       <div className="container">
         {Questions.map((el: IGame, index: number) => (
+          // Display a button with an arrow for each game from the Questions array
           <ListButtonWithArrow displayName={el.displayName} link={path + "/"+ el.name} key={index} />
-          ))}
-        <ListItem displayName="Больше книг-игр уже скоро!" />
+        ))}
+        <ListItem displayName="<Больше книг-игр уже скоро>!" />
```
Comment (dgusakov) [inline src/components/QuizMain.tsx:22]: What are these symbols for?

**Review by dgusakov — CHANGES_REQUESTED** [review]
Not a full review. Will finalise it later

File: src/components/App.tsx (hunk 1 above)
Comment (larmork) [reply src/components/App.tsx:46]: Forgot to delete

File: src/components/QuizMain.tsx (hunk 3 above)
Comment (larmork) [reply src/components/QuizMain.tsx:22]: Idk, just typo

File: src/components/Question.tsx (hunk 2 above)
Comment (larmork) [reply src/components/Question.tsx:72]: So, if there are no questions to show, we'll display "start over" button?

File: src/css/App.css [hunk 4]
```diff
@@ -128,6 +128,18 @@ html {
   color: #000000;
 }
 
+.button-1 {
+  display:block;
+  margin: 20px;
+  margin-left: 10px;
+}
+
+.button-2 {
```
Comment (dgusakov) [inline src/css/App.css:137]: Same as button-1

File: src/components/Question.tsx [hunk 5]
```diff
@@ -32,18 +89,41 @@ function Question(props: any) {
...
+              // If all questions have been shown, disable the button
+              <div>
+              <button
+                id="switchQuestion"
+                className="btn btn-scd btn-block"
+                type="button"
+                disabled
+              >
+                Все вопросы из этой темы уже показаны
+              </button>
```
Comment (dgusakov) [inline src/components/Question.tsx:106]: Keep only "Start over" button

File: src/components/Question.tsx (hunk 2 above)
Comment (dgusakov) [inline src/components/Question.tsx:72]: Change text to "Вопросов по теме не найденно"

**Review by dgusakov — CHANGES_REQUESTED** [review]

File: src/css/App.css [hunk 6]
```diff
@@ -128,6 +128,18 @@ html {
   color: #000000;
 }
 
+.button-1 {
```
[the remark this answers is not in the document]
Comment (larmork) [reply src/css/App.css]: fixed

File: src/css/App.css (hunk 4 above)
Comment (larmork) [reply src/css/App.css]: fixed

**Review by dgusakov — APPROVED** [review]

**Pull request merged**